spec.md

```diff
@@ -148,30 +148,30 @@ that contains both context and data).
 
 ### source
 * Type: Object
-* Description: This describes the software instance that emits the event at
-  runtime (i.e. the producer). It contains sub-properties (listed below)
+* Description: This describes the software where an event occurred. It contains sub-properties (listed below)
 * Constraints:
   * REQUIRED
   * MUST contain at least one non-empty sub-property.
 
 ### source-type
 * Type: String
-* Description: Type of the event source. Providers define list of event
-  sources.
+* Description: The kind of object on which an an event occurred. Providers define list of source-types.
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
 * Examples:
   * s3
+  * "google.firestore.document"
 
 ### source-id
 * Type: String
-* Description: ID of the event source.
+* Description: A unique identifier of the object on which an event occurred. The source-id is of kind source-type.
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
```

Comment:
I think we should add a statement in this Constraints section about the uniqueness aspect - is it globally unique or just unique within the scope of the source system?

Reply:
`object` may be ambiguous? e.g. s3 is not an object but a service.

Reply:
Honestly I struggled here a bit and decided to toss it back for discussion. If we add `source-server`, then it's very easy to say that `source-server` is a physical or logical address and `source-id` is a unique resource hosted by `source-server`.

As it is, I can't actually say that this is a globally-unique value. 10.x.x.x IP addresses are totally valid but may not be _globally_ unique.